src/storage/v2/indices/text_index.cpp

```diff
@@ -0,0 +1,470 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "storage/v2/indices/text_index.hpp"
+#include "flags/run_time_configurable.hpp"
+#include "query/db_accessor.hpp"
+#include "storage/v2/view.hpp"
+#include "text_search.hpp"
+
+namespace memgraph::storage {
+
+void TextIndex::AddNode(Vertex *vertex_after_update, Storage *storage, const std::uint64_t transaction_start_timestamp,
```

Comment:
Buda and I have considered using property IDs instead of their names, but we’ve decided to use property names (see [Notion](https://www.notion.so/memgraph/Text-search-basics-R-D-Phase-1-dbec52d4fdaa4cc0b0ccc74be1e1c573?p=5a186d0a0e1546c783ce7a31870ce3e8&pm=s)).
If we used property IDs, it would be impossible for users to search for given properties, e.g. `text_search.search("myIndex", "data:someProperty:text")`.

Reply:
it should be possible if you find property ID using NameIdMapper when you perform search... User specifies string, we are smart and serialize just ID. When user searches, it finds corresponding ID. But OK you can leave this as it is but I still don't like Storage here because you don't need storage here

Reply:
* `NameIdMapper`: It’s a better choice than `Storage` so I’ll remove storage from the method signature
* Finding property IDs: Not feasible, since we’d have to a) parse the Tantivy query language string b) map the property names to IDs c) pass the modified query to Tantivy